chore(docs): Update the "Gatsby on Linux" page (#18263)
gatsbyjs/gatsby: docs/docs/gatsby-on-linux.md

title: Gatsby on Linux
---

> # Linux

Reviewer: We can omit this, as it's duplicating the page header. That way, there will be only one h1 per page (the frontmatter title generates one already).
```suggestion
```

> This guide assumes you already have a native installation of Linux on your machine. The following steps walk through how to install `node` and associated depdendencies.

Reviewer: ```suggestion
This guide assumes you already have a native installation of Linux on your machine. The following steps walk through how to install Node.js and associated dependencies.
```

> ### Ubuntu, Debian and other `apt` based distros

Reviewer: This should be an h2, since it follows the h1.
```suggestion
## Ubuntu, Debian, and other `apt` based distros
```

Begin by updating and upgrading.

```shell
sudo apt update
sudo apt -y upgrade
```

Install curl which allows you to transfer data and download additional dependencies.

```shell
sudo apt install curl
```

Once `curl` is installed, you can use it to install `nvm`, which will manage `node` and all its associated versions.

```shell
curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.35.1/install.sh | bash
```

> Note that this is the current stable release of nvm. Full installation instructions and troubleshooting can be found at the [nvm GitHub page](https://github.com/nvm-sh/nvm)

> When `nvm` is installed, it does not default to a particular `node` version. You'll need to install the version you want and give `nvm` instructions to use it. This example uses the latest release of version `10`, but other version numbers can be used instead.

Reviewer: ```suggestion
When `nvm` is installed, it does not default to a particular `node` version. You'll need to install the version you want and give `nvm` instructions to use it. This example uses the latest release of version `10`, but more recent version numbers can be used instead.
```

```shell
nvm install 10
nvm use 10
```

To confirm this has worked, use the following command.

```shell
node -v
```

> Note that `npm` comes packaged with `node`

Finally, install `git` which will be necessary for creating your first Gatsby project based on a starter.

```shell
sudo apt install git
```

> ### Fedora, RedHat and other `dnf` based distros

Reviewer: ```suggestion
## Fedora, RedHat, and other `dnf` based distros
```

These distros come installed with `curl`, so you can use that to download `nvm`.

```shell
curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.35.1/install.sh | bash
```

> Note that this is the current stable release of nvm. Full installation instructions and troubleshooting can be found at the [nvm GitHub page](https://github.com/nvm-sh/nvm)

> When `nvm` is installed, it does not default to a particular `node` version. You'll need to install the version you want and give `nvm` instructions to use it. This example uses the latest release of version `10`, but other version numbers can be used instead.

Reviewer: ```suggestion
When `nvm` is installed, it does not default to a particular `node` version. You'll need to install the version you want and give `nvm` instructions to use it. This example uses the latest release of version `10`, but more recent version numbers can be used instead.
```

```shell
nvm install 10
nvm use 10
```

To confirm this has worked, use the following command.

```shell
node -v
```

> Note that `npm` comes packaged with `node`

Finally, install `git` which will be necessary for creating your first Gatsby project based on a starter.

```shell
sudo dnf install git
```

> ### Archlinux and other `pacman` based distros

Reviewer: ```suggestion
## Archlinux and other `pacman` based distros
```

Begin by updating.

```shell
sudo pacman -Sy
```

These distros come installed with `curl`, so you can use that to download `nvm`.

```shell
curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.35.1/install.sh | bash
```

> Note that this is the current stable release of nvm. Full installation instructions and troubleshooting can be found at the [nvm GitHub page](https://github.com/nvm-sh/nvm)

Before using `nvm`, you need to install additional dependencies.

```shell
sudo pacman -S grep awk tar git
```

> When `nvm` is installed, it does not default to a particular `node` version. You'll need to install the version you want and give `nvm` instructions to use it. This example uses the latest release of version `10`, but other version numbers can be used instead.

Reviewer: ```suggestion
When `nvm` is installed, it does not default to a particular `node` version. You'll need to install the version you want and give `nvm` instructions to use it. This example uses the latest release of version `10`, but more recent version numbers can be used instead.
```

```shell
nvm install 10
nvm use 10
```

To confirm this has worked, use the following command.

```shell
node -v
```

> Note that `npm` comes packaged with `node`

## Windows Subsystem Linux (WSL)

> This guide assumes that you already have WSL installed with a working Linux distro. If you don't, follow [this guide from microsoft's site](https://docs.microsoft.com/en-us/windows/wsl/install-win10) to install WSL and a Linux distro of your choice.

Reviewer: ```suggestion
This guide assumes that you already have WSL installed with a working Linux distro. If you don't, follow [this guide from Microsoft's site](https://docs.microsoft.com/en-us/windows/wsl/install-win10) to install WSL and a Linux distro of your choice.
```

title: Gatsby on Linux
---

# Linux

This guide assumes you already have a native installation of Linux on your machine. The following steps walk through how to install `node` and associated depdendencies.

### Ubuntu, Debian and other `apt` based distros

Begin by updating and upgrading.

```shell
sudo apt update
sudo apt -y upgrade
```

> > Only use `-y` if you want to upgrade to the latest version of the software.

Reviewer 2: The `-y` iirc will just automatically move forward w/o asking user for confirmation.

Reviewer 3: I stole that from below, but good point on fixing the wording and being clearer.

Install curl which allows you to transfer data and download additional dependencies.

```shell
sudo apt install curl
```

Once `curl` is installed, you can use it to install `nvm`, which will manage `node` and all its associated versions.

```shell
curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.35.1/install.sh | bash
```

> Note that this is the current stable release of nvm. Full installation instructions and troubleshooting can be found at the [nvm GitHub page](https://github.com/nvm-sh/nvm)

> When `nvm` is installed, it does not default to a particular `node` version. You'll need to install the version you want and give `nvm` instructions to use it. This example uses `10.17.0`, but other version numbers can be used instead.

Reviewer 2: IIRC with `nvm` you can just ask it to install the major version, and it will automatically pick up the latest release, i.e. `nvm install 10`

Reviewer 2: This way we don't have to update the materials when a new minor/patch comes out

Finally, install `git` which will be necessary for creating your first Gatsby project based on a starter.

```shell
sudo apt install git
```

### Fedora, RedHat and other `dnf` based distros

These distros come installed with `curl`, so you can use that to download `nvm`.

```shell
curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.35.1/install.sh | bash
```

> Note that this is the current stable release of nvm. Full installation instructions and troubleshooting can be found at the [nvm GitHub page](https://github.com/nvm-sh/nvm)

When `nvm` is installed, it does not default to a particular `node` version. You'll need to install the version you want and give `nvm` instructions to use it. This example uses `10.17.0`, but other version numbers can be used instead.

```shell
nvm install 10.17.0
nvm use 10.17.0
```

To confirm this has worked, use the following command.

```shell
node -v
```

> Note that `npm` comes packaged with `node`

Finally, install `git` which will be necessary for creating your first Gatsby project based on a starter.

```shell
sudo dnf install git
```

### Archlinux and other `pacman` based distros

Begin by updating.

```shell

> sudo pacman -Sy

Reviewer 2: @nicholascapo do you recall if we need this as a separate step for arch

Reviewer 3: It was needed when I ran it in docker fwiw

Reviewer 2: 👍

# Linux

> 

Reviewer 4: Pls, undo these changes, as WSL !== Linux. The idea is the same but the underlying things may differ a bit.

Author: I know it's different, it was just mentioned in #15529 for the "TODO" part to be removed, I will add it again. I have only made some small changes in this file.

## Windows Subsystem Linux (WSL)

> This guide assumes that you already have WSL installed with an working Linux distro. If you don't, follow [this guide from microsoft's site](https://docs.microsoft.com/en-us/windows/wsl/install-win10) to install WSL and a Linux distro of your choice.

Reviewer 5: > with an working Linux
Should be 
with a working Linux

> This guide assumes that you already have WSL installed with an working Linux distro. If you don't, follow [this guide from microsoft's site](https://docs.microsoft.com/en-us/windows/wsl/install-win10) to install WSL and a Linux distro of your choice.

Reviewer 3: ```suggestion
This guide assumes that you already have WSL installed with a working Linux distro. If you don't, follow [this guide from microsoft's site](https://docs.microsoft.com/en-us/windows/wsl/install-win10) to install WSL and a Linux distro of your choice.
```

> sudo apt update && sudo apt -y upgrade && sudo apt install build-essential && sudo apt install git && sudo apt install libpng-dev

Reviewer 4: Nice work, can you write all commands separately with the proper description, so that users know about what the command will do on execution.

Author: Sure, and another thing I was wondering is, I have installed gatsby a couple times on both windows and linux, and never had to install build-essential (on linux) or windows-build-tools (on windows), but in the docs it says you have to install them, are they really necessary?

Reviewer 6: I think you do need them for gatsby-plugin-sharp and probably some other plugins. I'm unsure if gatsby core needs it.

Reviewer 5: `sharp` ships with binaries for windows, no build tools required.

Reviewer 5: http://sharp.pixelplumbing.com/en/stable/install/

Reviewer 5: Sharp tries to use supplied binaries first but will sometimes fall back to compiling source. http://sharp.pixelplumbing.com/en/stable/install/

Author: Marking as resolved since @laurieontech fixed this.

> sudo apt update && sudo apt -y upgrade && sudo apt install build-essential && sudo apt install git && sudo apt install libpng-dev

Reviewer 4: These commands are specific to Debian distros, it would be great if you provide similar docs for other distros like cent, arch, etc.

Author: I have never used them but I will look them up.

Author: Marking as resolved since @laurieontech fixed this.

```

### Additional links and resources

> - [Super detailed guide to making VSCode work with ESL from VSCode's docs website](https://code.visualstudio.com/docs/remote/wsl)

Reviewer 6: Small typo

```suggestion
- [Super detailed guide to making VSCode work with WSL from VSCode's docs website](https://code.visualstudio.com/docs/remote/wsl)
```

Author: Yes! I will add more detailed instructions while adding instructions to the part 0 page! I could split them by package manager too, like mentioned there.

- [windows store](https://www.microsoft.com/en-us/store/p/ubuntu/9nblggh4msv6)
- [n](https://github.com/tj/n)
- [n-install](https://github.com/mklement0/n-install)
- [nvm](https://github.com/creationix/nvm)

> - [bash startup](https://github.com/Microsoft/WSL/issues/776#issuecomment-266112578)

Reviewer 4: One line description would be helpful before visiting. As the reader can pick what is useful or not for his/her context.

Author: Hmmm since they are all mentioned in the article, I could explain them there. I just enabled the links!
